public/app/features/variables/editor/VariableEditorEditor.tsx

```diff
@@ -110,104 +119,90 @@ export class VariableEditorEditorUnConnected extends PureComponent<Props> {
+      { label: '', value: VariableHide.dontHide },
+      { label: 'Label', value: VariableHide.hideLabel },
+      { label: 'Variable', value: VariableHide.hideVariable },
+    ];
+    const hideValue = hideOptions.find(o => o.value === variable.hide) ?? hideOptions[0];
 
     return (
       <div>
         <form aria-label="Variable editor Form" onSubmit={this.onHandleSubmit}>
-          <h5 className="section-heading">General</h5>
-          <div className="gf-form-group">
-            <div className="gf-form-inline">
-              <div className="gf-form max-width-19">
-                <span className="gf-form-label width-6">Name</span>
-                <input
-                  type="text"
-                  className="gf-form-input"
-                  name="name"
-                  placeholder="name"
-                  required
+          <VerticalGroup spacing="lg">
+            <VerticalGroup spacing="none">
+              <VariableSectionHeader name="General" />
+              <InlineFieldRow>
+                <VariableTextField
                   value={this.props.editor.name}
                   onChange={this.onNameChange}
-                  aria-label={selectors.pages.Dashboard.Settings.Variables.Edit.General.generalNameInput}
+                  name="Name"
+                  placeholder="name"
+                  required
+                  ariaLabel={selectors.pages.Dashboard.Settings.Variables.Edit.General.generalNameInput}
+                />
+                <VariableSelectField
+                  name="Type"
+                  value={typeValue}
+                  options={typeOptions}
+                  onChange={this.onTypeChange}
+                  tooltip={typeTooltip}
+                  ariaLabel={selectors.pages.Dashboard.Settings.Variables.Edit.General.generalTypeSelect}
                 />
-              </div>
-              <div className="gf-form max-width-19">
-                <InlineFormLabel width={6} tooltip={variableAdapters.get(this.props.variable.type).description}>
-                  Type
-                </InlineFormLabel>
-                <div className="gf-form-select-wrapper max-width-17">
-                  <select
-                    className="gf-form-input"
-                    value={this.props.variable.type}
-                    onChange={this.onTypeChange}
-                    aria-label={selectors.pages.Dashboard.Settings.Variables.Edit.General.generalTypeSelect}
-                  >
-                    {getVariableTypes().map(({ label, value }) => (
-                      <option key={value} label={label} value={value}>
-                        {name}
-                      </option>
-                    ))}
-                  </select>
+              </InlineFieldRow>
+
+              {this.props.editor.errors.name && (
+                <div className="gf-form">
+                  <span className="gf-form-label gf-form-label--error">{this.props.editor.errors.name}</span>
```

Comment:
nit: do we have any components for this one?

Reply:
We could probably use the react form hook instead because that should have it's own error handling. I haven't used it before so I'll look into that.